homeassistant/components/whirlpool/climate.py

```diff
@@ -110,8 +110,20 @@ def __init__(self, hass, said, name, backend_selector: BackendSelector, auth: Au
         self._aircon = Aircon(backend_selector, auth, said, self.async_write_ha_state)
 
         self.entity_id = generate_entity_id(ENTITY_ID_FORMAT, said, hass=hass)
-        self._attr_name = name if name is not None else said
+        self._name = name if name is not None else said
         self._attr_unique_id = said
+        self._said = said
+        self._attr_has_entity_name = True
```

Comment:
I would make this a class constant since this will always be True.

Reply:
I don't think I can move _said to a class constant - the value is passed in.